### PR TITLE
fix: fix materials panel

### PR DESCRIPTION
**Pull request by yy-wow**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?
1. 物料面板固定面板按钮点击无效
2. 固定面板下次进入需要再次固定，没有持久化保存

Issue Number: N/A

### What is the new behavior?
1. 物料面板固定面板按钮点击固定或解除固定正常
2. 持久化保存固定状态，提升体验

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced header interactivity by adding an event emission for the `fix-panel` action, allowing better communication with parent components.
	- Introduced a computed property, `panelFixed`, to streamline the logic for determining the fixed state of panels in the header and tree components.
	- Implemented local storage functionality to remember fixed panels across sessions, ensuring user preferences persist after page refreshes.
- **Improvements**
	- Cleaned up the code by removing unused sections and simplifying the logic for panel state representation.
	- Added error handling for local storage operations to enhance user experience.

### CODE REVIEW

**Review by chilingling — DISMISSED** [review]

File: packages/plugins/materials/src/meta/layout/src/Main.vue [hunk 1]
```diff
@@ -1,7 +1,11 @@
 <template>
   <plugin-panel :title="shortcut ? '' : title" @close="$emit('close')">
     <template #header>
-      <component :is="headerComponent" :fixedPanels="fixedPanels"></component>
+      <component
+        :is="headerComponent"
+        :fixedPanels="fixedPanels"
+        @fix-panel="(id) => $emit('fix-panel', id)"
```
Comment (hexqi) [inline packages/plugins/materials/src/meta/layout/src/Main.vue:7]: 点击固定后图标状态没有切换，📌应该切换为钉住状态
image

Comment (yy-wow) [reply packages/plugins/materials/src/meta/layout/src/Main.vue:7]: 已添加新图标

File: packages/design-core/assets/fixed-solid.svg [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<svg width="16.000000" height="16.000000" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"
+	xmlns:xlink="http://www.w3.org/1999/xlink">
+	<desc>
+		Created with Pixso.
```
Comment (chilingling) [inline packages/design-core/assets/fixed-solid.svg]: 没用的注释干掉

File: packages/layout/src/DesignPlugins.vue [hunk 3]
```diff
@@ -154,8 +157,30 @@ export default {
...
+        if (!Array.isArray(pluginState.fixedPanels)) {
+          pluginState.fixedPanels = []
+        }
+      } catch (error) {
+        message({ message: `'Failed to restore fixed panels state:'${error}`, status: 'error' })
+        pluginState.fixedPanels = []
+      }
+    }
+
+    restoreFixedPanels()
```
Comment (chilingling) [inline packages/layout/src/DesignPlugins.vue:183]: 放在 `onMounted` 中会不会好一点？

Comment (yy-wow) [reply packages/layout/src/DesignPlugins.vue:183]: 这里的数据不需要dom渲染后的状态，在上面已经拿到了pageState，直接赋值看上去没有影响

**Review by hexqi — APPROVED** [review]

**Review by chilingling — APPROVED** [review]

**Pull request merged**